tests/functional/states/test_kubernetes.py

```diff
@@ -0,0 +1,1426 @@
+                image: {{ image }}
+                ports:
+                - containerPort: 80
+        """
+    ).strip()
+
+    with pytest.helpers.temp_file(f"{sls}.yml.jinja", contents, state_tree):
+        yield f"salt://{sls}.yml.jinja"
+
+
+@pytest.fixture(params=[True])
+def deployment(kubernetes, _deployment_spec, request):
+    """
+    Fixture providing a deployment for testing
+    """
+    name = random_string("deployment-", uppercase=False)
+    namespace = "default"
+
+    # Only create the deployment if requested
+    if request.param:
+        ret = kubernetes.deployment_present(
+            name=name,
+            namespace=namespace,
+            spec=_deployment_spec,
+            wait=True,
+        )
+        assert ret.result is True
+    try:
+        yield {"name": name, "namespace": namespace}
+    finally:
+        ret = kubernetes.deployment_absent(name=name, namespace=namespace, wait=True)
+        assert ret.result is True
+        assert ret.comment in ["The deployment does not exist", "None"]
+
+
+@pytest.mark.parametrize("deployment", [False], indirect=True)
+def test_deployment_present(kubernetes, deployment, _deployment_spec):
+    """
+    Test kubernetes.deployment_present creates a deployment
+    """
+    ret = kubernetes.deployment_present(
+        name=deployment["name"],
+        namespace=deployment["namespace"],
+        spec=_deployment_spec,
+        wait=True,
+    )
+
+    assert ret.result is True
+    assert ret.changes
+
+    # Comment out idempotent test for now
+    # TODO: The state module needs fixed to handle proper present functionality
+
+
+# def test_deployment_present_idempotency(kubernetes, deployment, _deployment_spec):
+#     """
+#     Test kubernetes.deployment_present is idempotent
+#     """
+#     ret = kubernetes.deployment_present(
+#         name=deployment["name"],
+#         namespace=deployment["namespace"],
+#         spec=_deployment_spec,
+#         wait=True,
+#     )
+#     assert ret.result is True
+#     assert "already exists" in ret.comment
+#     assert not ret.changes
+
+
+def test_deployment_present_replace(kubernetes, deployment):
+    """
+    Test kubernetes.deployment_present replaces a deployment
+    """
+    new_spec = {
+        "replicas": 3,
+        "selector": {"matchLabels": {"app": "test"}},
+        "template": {
+            "metadata": {"labels": {"app": "test"}},
+            "spec": {
+                "containers": [
+                    {
+                        "name": "nginx",
+                        "image": "nginx:latest",
+                        "ports": [{"containerPort": 80}],
+                    }
+                ]
+            },
+        },
+    }
+
+    ret = kubernetes.deployment_present(
+        name=deployment["name"],
+        namespace=deployment["namespace"],
+        spec=new_spec,
+        wait=True,
+    )
+
+    assert ret.result is True
+    assert ret.changes["spec"]["replicas"] == 3
```

Comment:
```suggestion
def test_deployment_present_replace(kubernetes, deployment, _deployment_spec):
    """
    Test kubernetes.deployment_present replaces a deployment
    """
    _deployment_spec["replicas"] = 3

    ret = kubernetes.deployment_present(
        name=deployment["name"],
        namespace=deployment["namespace"],
        spec=_deployment_spec,
        wait=True,
    )

    assert ret.result is True
    assert ret.changes["spec"]["replicas"] == 3
```

suggestion: Keep it DRY by reusing the `_deployment_spec` fixture.